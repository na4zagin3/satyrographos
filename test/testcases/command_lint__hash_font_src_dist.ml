open Core
open Satyrographos_testlib

let satysfi_package_opam =
  "satysfi-package.opam", TestLib.opam_file_for_test
    ~name:"satysfi-package"
    ~version:"0.1"
    ()

let satyristes =
  "Satyristes", sprintf
    {|(version "0.0.2")

(library
  (name "package")
  (version "0.1")
  (sources ((hash "mathfonts.satysfi-hash" "mathfonts.satysfi-hash")
            (hash "fonts.satysfi-hash" "fonts.satysfi-hash")))
  (opam "satysfi-package.opam")
  (dependencies ()))
|}

let fonts_satysfi_hash =
  "fonts.satysfi-hash", {|
{
  "font1": <Single:{"src-dist": "abc.ttf" }>
}
|}

let mathfonts_satysfi_hash =
  "mathfonts.satysfi-hash", {|
{
  "mathfont1": <Single:{"src-dist": "math-abc.ttf" }>
}
|}

let opam_libs = Satyrographos.Library.[
    {empty with
     name = Some "package";
     files = LibraryFiles.of_alist_exn [
         "packages/package/test.satyh", `Content ""
       ]
    };
]

let files =
  [ satysfi_package_opam;
    satyristes;
    fonts_satysfi_hash;
    mathfonts_satysfi_hash;
  ]

let () =
  TestCommand.test_lint_command ~opam_libs files
