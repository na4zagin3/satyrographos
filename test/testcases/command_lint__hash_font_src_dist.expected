@@test_library@@/Satyristes:3:1: (module package):
Warning: hash/font/location/src-dist
  src-dist field in font hashes (font1) has been deprecated.  See https://github.com/gfngfn/SATySFi/commit/868fd678aaa19aeb59d4fb77c928e833af4c2359

@@test_library@@/Satyristes:3:1: (module package):
Warning: hash/font/location/src-dist
  src-dist field in font hashes (mathfont1) has been deprecated.  See https://github.com/gfngfn/SATySFi/commit/868fd678aaa19aeb59d4fb77c928e833af4c2359

2 problem(s) found.
